http_test.go

```diff
@@ -125,10 +125,15 @@ func TestIterator(t *testing.T) {
 						errReq = reqs[0]
 					}
 
-					httpSvc := svc.HTTP.RateLimiter(rlimiter).Requests(reqs...)
-					httpSvc.client = newMockHTTPClient(
-						withMockHTTPClientResponseError(errReq, tcase.forceErr),
-						withMockHTTPClientRequests(reqs...))
+					httpSvc := NewHTTPService(
+						svc,
+						// options
+						WithRateLimiter(rlimiter),
+						WithRequests(reqs...),
+						WithClient(newMockHTTPClient(
+							withMockHTTPClientResponseError(errReq, tcase.forceErr),
+							withMockHTTPClientRequests(reqs...)),
+						))
 
```

Comment:
I wasnt sure how best to format this code but this way looked the cleanest to me. Let me know what you think and if youd like any changes. @prestonvasquez